### DISCUSSION TITLE
SHAP values for regression Problems?

### DISCUSSION THREAD

**Discussion by ockaro** [shap/shap#3026] [General]
Hi there! :) I just wanted to know if there is a possibility to compute SHAP values for multi-output regression problems which are modeled using a neural network. Looking at this example https://slundberg.github.io/shap/notebooks/NHANES%20I%20Survival%20Model.html it generaly seems to work but maybe only for tree based models? Because when I tried to generate shap values for my problem I only got zeros while the same code works perfectly for a different classification task of mine.

**Comment by jainraj** [comment]
What Explainer are you using? For NNs, my personal recommendation is GradientExplainer.

**Reply by ockaro** [reply]
Hi @jainraj , I am also using the GradientExplainer, but when I call it for a NN which solves a regression problem I get empty plots. Additionally I could not find an example for that case so I asked myself if it is possible?

**Reply by jainraj** [reply]
Mind sharing a simplified example code? Specifically these: 

1. simplified NN code 
1. applying GradientExplainer to your NN
2. plotting code 

And at least from reading the GradientExplainer code, it seems like multi-output NNs are supported.
